lib/scanner.js

```diff
@@ -60,6 +54,32 @@ class Scanner {
   rewind (pos) {
     this.pos = pos
   }
+
+  enter (type, content) {
+    const position = { start: this.position() }
+    return Array.isArray(content)
+      ? { type, children: content, position }
+      : { type, value: content, position }
+  }
+
+  exit (node) {
+    node.position.end = this.position()
+    return node
+  }
+
+  abort (node, expectedTokens) {
+    const position = `${this.pos.line}:${this.pos.column}`
```

Comment:
We can also use `unist-util-stringify-position` instead of the `${pos.line}:${pos.column}`, because we have the attempted node information?

Reply:
I feel like there will be tooling value in having this as as well `{ line: <line>, column: <column> }` on the error.  Maybe before line 80 we could add `error.position = this.pos`?

Reply:
Another helpful pattern I like to follow (used in node, which is good precedence I think) is to have distinct `.code` properties for these errors.  If we were tooling around errors, checking `err.code === 'EOF_ERROR` is much better than `err.message.startsWith('unexpected token EOF at')`.

Reply:
I agree there would be value in adding additional context to the error object. I also wonder if we should consider returning an error rather than ever throwing? or perhaps have a "best effort" mode, and a mode that throws?